pkg/plugins/gateway/cloudrun/http.go

```diff
@@ -47,24 +47,36 @@ func middleware(ctx *fasthttp.RequestCtx, pool worker.WorkerPool) bool {
 	if err := json.Unmarshal(bodyBytes, &pubsubEvent); err == nil && pubsubEvent.Subscription != "" {
 		// We have an event from pubsub here...
 
+		topicName, ok := pubsubEvent.Message.Attributes["x-nitric-topic"]
+		if !ok {
+			ctx.Error("Could not find topicName for event", 400)
+			return false
+		}
+
 		// need to determine if the underlying data is a nitric event
 		var event *triggers.Event
 		messageJson := &ep.NitricEvent{}
+
 		// Check if it's a nitric event
-		if err := json.Unmarshal(pubsubEvent.Message.Data, messageJson); err == nil && messageJson.ID != "" {
+		err := json.Unmarshal(pubsubEvent.Message.Data, messageJson)
+		if err == nil && messageJson.ID != "" && messageJson.PayloadType != "" {
```

Comment:
payload type in some cases may be optional (we don't actually enforce its delivery)

Reply:
ok, what else do you suggest. because almost all objects have an "id". so that is really not a good indication of it being a nitric event.

Reply:
At the moment I wouldn't apply any changes. Currently looking into using the cloud events spec for our eventing and would end up deprecating the "Nitric Event" anyway.